packages/ui/src/components/Metrics/components/ChartPartitionSizePerBroker.tsx

```diff
@@ -0,0 +1,190 @@
+import type { ChartVoronoiContainerProps } from "@patternfly/react-charts";
+import { ChartLine } from "@patternfly/react-charts";
+import {
+  Chart,
+  ChartAxis,
+  ChartGroup,
+  ChartLegend,
+  ChartThemeColor,
+  ChartVoronoiContainer,
+} from "@patternfly/react-charts";
+import {
+  chart_color_blue_300,
+  chart_color_cyan_300,
+} from "@patternfly/react-tokens";
+import type { FunctionComponent, ReactElement } from "react";
+import { useTranslation } from "react-i18next";
+import { chartHeight, chartPadding } from "../consts";
+import type { PartitionBytesMetric, PartitionSelect } from "../types";
+import { ChartSkeletonLoader } from "./ChartSkeletonLoader";
+import { useChartWidth } from "./useChartWidth";
+import {
+  dateToChartValue,
+  formatBytes,
+  shouldShowDate,
+  timestampsToTicks,
+} from "./utils";
+
+const colors = [chart_color_cyan_300.value, chart_color_blue_300.value];
+
+type ChartData = {
+  color: string;
+  area: PartitionChartData[];
+};
+
+type PartitionChartData = {
+  name: string;
+  x: number;
+  y: number;
+};
+
+type LegendData = {
+  name: string;
+};
+
+export type ChartPartitionSizePerBrokerProps = {
+  partitions: PartitionBytesMetric;
+  broker: string | undefined;
+  duration: number;
+  isLoading: boolean;
+  emptyState: ReactElement;
+  selectedPartition: PartitionSelect;
+};
+export const ChartPartitionSizePerBroker: FunctionComponent<
+  ChartPartitionSizePerBrokerProps
+> = ({
+  partitions,
+  broker,
+  duration,
+  isLoading,
+  emptyState,
+  selectedPartition,
+}) => {
+    const { t } = useTranslation();
+    const [containerRef, width] = useChartWidth();
+
+    const { chartData, legendData, tickValues } = getChartData(
+      partitions,
+      broker,
+      duration,
+      selectedPartition
+    );
+
+    const hasMetrics = Object.keys(partitions).length > 0;
+
+    const showDate = shouldShowDate(duration);
+
+    return (
+      <div ref={containerRef} style={{ marginTop: "-30px", height: "700px" }}>
```

Comment:
Hi @jgiardino, I have fixed the legend name visible issue by adjusting the height and 
 [This](https://63c984f3273772562ae72406-ymjdmvpxrf.chromatic.com/?path=/story/components-metrics--all-ready ) is how the charts look, could you please have a look, if the topic length is more again some of the names of the partitions may not be visible, please let me know your opinion on this